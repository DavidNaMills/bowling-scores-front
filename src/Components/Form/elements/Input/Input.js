import React from 'react';
import classes from './Input.module.scss';
import { CreateSharp, ErrorSharp } from '@material-ui/icons'

const iconOk = [classes.icon, classes.icon__required];
const iconErr = [classes.icon, classes.icon__error];

const Input = ({ changed, id, type = 'text', label = null, placeholder = '', error = [], isValid = false, isRequired = false }) => {
    let styles = [classes.input, classes.input__default]

    if (isValid) {
        styles = styles.concat(classes.input__valid);
    }

    if (error.length > 0) {
        styles = styles.concat(classes.input__error);
    }

    return (
        <div className={classes.input__label}>
            {label && <label htmlFor={id}>{label}</label>}
            <div>
                <input
                    className={styles.join(' ')}
                    id={id}
                    type={type}
                    onChange={e => changed(e, id)}
                    placeholder={placeholder}
                />

                {
                    error.length > 0
                        ? <ErrorSharp className={iconErr.join(' ')} />
                        : isRequired && <CreateSharp className={iconOk.join(' ')} />
                }
            </div>
            {error.length > 0 &&
                <div className={classes.input__errorContainer}>
                    {
                        error.map((x, i) => <p key={i} className={classes.input__errorContainer_errorMsg}>{x}</p>, 0)
                    }
                </div>
            }
        </div>
    )
}

export default Input;