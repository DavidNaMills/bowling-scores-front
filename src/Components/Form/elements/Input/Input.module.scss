@import '../../../../styles/colors.scss';
@import '../../../../styles/fonts.scss';

.input{
    border: none;
    padding: 5px;
    width: $input_width;
    
    &__label{
        margin-top: $input_spacing;
    }

    &__default{
        background: rgb(230, 230, 230);
    }

    &__valid{
        -webkit-box-shadow: $valid_input_shadow;
        -moz-box-shadow: $valid_input_shadow;
        box-shadow: $valid_input_shadow;
        border-bottom: solid 1px $btn_default_bkg;
    };
    
    &__error{
        -webkit-box-shadow: $error_input_shadow;
        -moz-box-shadow: $error_input_shadow;
        box-shadow: $error_input_shadow;
        border-bottom: solid 1px $btn_danger_bkg;
    }

    &__errorContainer{
        width: $input_width;
        padding: 5px;

        -webkit-box-shadow: $error_input_shadow;
        -moz-box-shadow: $error_input_shadow;
        box-shadow: $error_input_shadow;

        &_errorMsg{
            font-size: $error_size_small;
            color: $btn_danger_bkg;
        }
    }
}

.icon{
    margin-left: 5px;
    display: flex;
    justify-content: center;
    align-content: center;

    &__required{
        color: $btn_default_bkg_active_hover;
    }

    &__error{
        color: $btn_danger_bkg_active;
    }
}