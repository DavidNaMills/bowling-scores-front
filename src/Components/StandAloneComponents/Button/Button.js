import React from 'react';
import classes from './Button.module.scss';



const Button = ({ click = () => { }, label, isDisabled = false, type = 'default' }) => {
    let style = [classes.button];

    if (isDisabled) {
        style = style.concat(classes.button__disabled);
    } else {

        switch (type.toLowerCase()) {
            case 'warning':
                style = style.concat(classes.button__warning);
                break;
            case 'danger':
                style = style.concat(classes.button__danger);
                break;
            default:
                style = style.concat(classes.button__default);
                break;
        }
    }

    return (
        <button onClick={click} disabled={isDisabled} className={style.join(' ')}>{label}</button>
    );
};

export default Button;