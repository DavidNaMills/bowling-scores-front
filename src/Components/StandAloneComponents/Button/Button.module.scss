@import '../../../styles/colors.scss';


.button{
    border: none;
    font-size: 16px;
    padding: 10px;
    min-width: 100px;
    font-weight: 550;

    &__disabled{
        background-color: $btn_disabled;
    }
    
    &__isFull{
        width: 100%;
    }

    &__default{
        background-color: $btn_default_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_default_bkg_active_hover;
        }
        &:active{
            background-color: $btn_default_bkg_active;
        }
    }

    &__warning{
        background-color: $btn_warning_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_warning_bkg_active_hover;
        }
        &:active{
            background-color: $btn_warning_bkg_active;
        }    
    }
    
    &__danger{
        background-color: $btn_danger_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_danger_bkg_active_hover;
        }
        &:active{
            background-color: $btn_danger_bkg_active;
        }    
    }

    &__lightGreen{
        background-color: $btn_lightGreen_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightGreen_bkg_active;
        }
        &:active{
            background-color: $btn_lightGreen_bkg_active_hover;
        }    
    }

    &__darkGreen{
        background-color: $btn_darkGreen_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkGreen_bkg_active;
        }
        &:active{
            background-color: $btn_darkGreen_bkg_active_hover;
        }    
    }

    &__blue{
        background-color: $btn_blue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_blue_bkg_active;
        }
        &:active{
            background-color: $btn_blue_bkg_active_hover;
        }    
    }

    &__lightBlue{
        background-color: $btn_lightBlue_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightBlue_bkg_active;
        }
        &:active{
            background-color: $btn_lightBlue_bkg_active_hover;
        }    
    }

    &__darkBlue{
        background-color: $btn_darkBlue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkBlue_bkg_active;
        }
        &:active{
            background-color: $btn_darkBlue_bkg_active_hover;
        }    
    }

    &__lightRed{
        background-color: $btn_lightRed_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightRed_bkg_active;
        }
        &:active{
            background-color: $btn_lightRed_bkg_active_hover;
        }    
    }

    &__lightYellow{
        background-color: $btn_lightYellow_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightYellow_bkg_active;
        }
        &:active{
            background-color: $btn_lightYellow_bkg_active_hover;
        }    
    }



}