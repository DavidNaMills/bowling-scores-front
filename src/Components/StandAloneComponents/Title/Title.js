import React from 'react';
import classes from './Title.module.scss';

const Title = ({ label = '', ttlType = 'sub' }) => {
    let item;

    switch (ttlType.toLowerCase()) {
        case 'main':
            item = <h1 className={classes.title__main}>{label}</h1>
            break;
        case 'section':
            item = <h2 className={classes.title__section}>{label}</h2>
            break;
        case 'sub':
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
        default:
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
    }

    return (
        <div className={classes.title}>
            {item}
        </div>
    )
}

export default Title;