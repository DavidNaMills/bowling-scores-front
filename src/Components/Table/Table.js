import React from 'react';
import classes from './Table.module.scss';

const Table = ({ caption = '', tableClass = classes.table, data, showRowNum = false, selectRow = () => { }, ...rest }) => (
    <div>
        {caption && <p className={classes.caption}>{caption}</p>}
        <table className={tableClass}>
            <thead>
                {showRowNum && <th></th>}
                {data.headers.map((x, i) =>
                    <th key={i}>
                        {x}
                    </th>
                    , 0)}
            </thead>
            <tbody>
                {
                    data.rows.map((x, i) =>
                        <tr
                            key={i}
                            className={x.style && typeof (x.style.trow) === 'string' ? x.style.trow : ''}
                            style={x.style && typeof (x.style.trow) === 'object' ? x.style.trow : {}}
                            onClick={() => selectRow(x.id ? x.id : i + 1)}
                        >
                            {showRowNum && <td key={`${i}0`}>{i + 1}</td>}
                            {
                                x.values.map((y, o) =>
                                    <td key={`${i}-${o}`}>
                                        {y}
                                    </td>
                                    , 1)
                            }
                        </tr>
                    )
                }
            </tbody>
        </table>
    </div>
)

export default Table;

// const tHeaders = ['name', 'Average', 'Pinfall', 'Rank'];
// const tData = [
//     {
//         id: '',
//         values: ['david', 135, 2445, '+ 2'],
//         style: { tr: classes.testRowClass }  //optional
//     },
//     {
//         id: '5vd5f3d5cvx3cv5xz53xcz',
//         values: ['alan', 189, 2222, '-2'],
//         style: { trow: { color: 'red' } }  //optional
//     }
// ];