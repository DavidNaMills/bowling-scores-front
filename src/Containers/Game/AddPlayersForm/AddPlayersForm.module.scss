.addPlayers{
    &__halfBtn{
        width: 50%;
    }

    &__fullBtn{
        width: 100%;
    }

    &__inline{
        display: flex;
        justify-content: space-between;
    }
}