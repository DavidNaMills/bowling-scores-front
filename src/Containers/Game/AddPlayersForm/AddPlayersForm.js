import React, { useState, useEffect } from 'react';
// import { useSelector } from 'react-redux';
import Button from '../../../Components/StandAloneComponents/Button/Button';
import Table from '../../../Components/Table/Table';
import Input from '../../../Components/Form/elements/Input/Input';
import Title from '../../../Components/StandAloneComponents/Title/Title';

import { playerTableStyle } from '../playerTableStyle';
import tableParser from '../../../helpers/tableGamesDataModifier/tableGamesDataModifier';

import useAddPlayer from '../../../Hooks/useAddPlayer/useAddPlayer';
import useColorSelection from '../../../Hooks/useColorSelection/useColorSelection';

const tHeaders = ['Player', 'Average', 'Pinfall'];



const AddPlayersForm = ({ title, playerSelect, onClose, isNew = false, close = () => { } }) => {
    // const liveGame = useSelector(state => state.liveGame);
    const {
        liveGame,   //temp structure
        newPlayer,
        setName,    //only sets player name
        addColor,    //sets color
        addPlayer,
        commitGame,
    } = useAddPlayer(isNew);

    const { color, randomColor, showColorPickerComponent } = useColorSelection();
    const [showSelect, setShowSelect] = useState(false);


    useEffect(() => {
        if (color) {
            addColor(color);
        }
    }, [color]);


    return (
        <div>
            <Title label={title} ttlType='sub' />
            <Input
                value={newPlayer.name}
                id='name'
                label='Players Name'
                changed={setName}
            />

            {
                showSelect && showColorPickerComponent()

            }
            <Button label={showSelect ? 'Close' : 'Select Color'} click={() => setShowSelect(prev => !prev)} />
            <Button label='Random Colour' click={randomColor} />
            {newPlayer.color &&
                <div
                    style={{
                        marginTop: '10px',
                        marginBottom: '10px',
                        height: '30px',
                        textAlign: 'center',
                        ...playerTableStyle(newPlayer.color)
                    }}
                >{newPlayer.name}</div>}


            <Button label='Add Player' click={addPlayer} />
            {isNew && <Button label='Start Game' click={() => commitGame(close)} />}
            <Table
                data={{
                    headers: tHeaders,
                    rows: tableParser(liveGame)
                }}
                selectRow={playerSelect}
                showRowNum
                caption='Click player for more details'
            />
            <Button label='Close' click={onClose} />
        </div>
    )
}

export default AddPlayersForm;