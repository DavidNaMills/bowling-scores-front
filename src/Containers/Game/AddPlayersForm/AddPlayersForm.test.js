import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import AddPlayersForm from './AddPlayersForm';
import Button from '../../../Components/StandAloneComponents/Button/Button';
import Table from '../../../Components/Table/Table';
import Input from '../../../Components/Form/elements/Input/Input';
import Title from '../../../Components/StandAloneComponents/Title/Title';

jest.mock('react-redux');
import {useSelector} from 'react-redux';

const mockState = {
    games: {
        1: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 12 } },
        2: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 24 } },
        3: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 154 } },
    },
    players: {
        'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", color: "17, 127, 14", id: "debb53c3-ad2a-4093-b89b-2a5025416f6c" }
    }
};
const splSelMock = jest.fn();
const onCloseMock = jest.fn();

afterEach(()=>{
    jest.clearAllMocks();
});

describe('<AddPlayersForm /> test suite', ()=>{

    beforeEach(()=>{
        useSelector.mockImplementation(()=>(mockState));
    });
    
    describe('Snapshot tests', ()=>{
        test('matches default snapshot', () => {
            const component = renderer.create(<AddPlayersForm 
                title='Test' 
                playerSelect={splSelMock}
                onClose={onCloseMock}
            />);
            const app = component.toJSON();
            expect(app).toMatchSnapshot();
        });
    });

    describe('Correct components are rendered', ()=>{
        it('renders the correct components', ()=>{
            const wrapper = mount(<AddPlayersForm 
                title='Test' 
                playerSelect={splSelMock}
                onClose={onCloseMock}
            />);

            expect(wrapper.find(Title).length).toBe(1);
            expect(wrapper.find(Input).length).toBe(1);
            expect(wrapper.find(Button).length).toBe(4);
            expect(wrapper.find(Table).length).toBe(1);

        });

        it('renders the correct components when isNew', ()=>{
            const wrapper = mount(<AddPlayersForm 
                title='Test' 
                playerSelect={splSelMock}
                onClose={onCloseMock}
                isNew
            />);

            expect(wrapper.find(Title).length).toBe(1);
            expect(wrapper.find(Input).length).toBe(1);
            expect(wrapper.find(Button).length).toBe(5);
            expect(wrapper.find(Table).length).toBe(1);

        });
    });
});