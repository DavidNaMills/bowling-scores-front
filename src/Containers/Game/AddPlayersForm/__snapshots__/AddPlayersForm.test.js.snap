// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPlayersForm /> test suite Snapshot tests matches default snapshot 1`] = `
<div
  className="contentContainer"
>
  <div
    className="title"
  >
    <h3
      className="title__sub"
    >
      Test
    </h3>
  </div>
  <div
    className="form__container"
  >
    <div
      className="form__form"
    >
      <div
        className="input__label"
      >
        <label
          htmlFor="name"
        >
          Players Name
        </label>
        <div>
          <input
            className="input input__default"
            id="name"
            onChange={[Function]}
            placeholder=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="addPlayers__inline extra"
      >
        <div
          className="addPlayers__halfBtn"
        >
          <button
            className="button button__isFull button__lightBlue"
            disabled={false}
            onClick={[Function]}
          >
            Select Color
          </button>
        </div>
        <div
          className="addPlayers__halfBtn"
        >
          <button
            className="button button__isFull button__lightGreen"
            disabled={false}
            onClick={[Function]}
          >
            Random Colour
          </button>
        </div>
      </div>
      <div
        className="largeExtra"
      >
        <button
          className="button button__isFull button__disabled"
          disabled={true}
          onClick={[Function]}
        >
          Add Player
        </button>
      </div>
    </div>
  </div>
  <div
    className="largeExtra"
  >
    <div>
      <p
        className="caption"
      >
        Click player for more details
      </p>
      <table
        className="table"
      >
        <thead>
          <tr>
            <td />
            <td>
              Player
            </td>
            <td>
              Average
            </td>
            <td>
              Pinfall
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            className=""
            onClick={[Function]}
            style={
              Object {
                "background": "linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(17, 127, 14,1) 100%)",
              }
            }
          >
            <td>
              1
            </td>
            <td>
              gfdgfds
            </td>
            <td>
              63
            </td>
            <td>
              190
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="largeExtra"
  >
    <button
      className="button button__isFull button__lightRed"
      disabled={false}
      onClick={[MockFunction]}
    >
      Close
    </button>
  </div>
</div>
`;
